modules/kubernetes-addons/opentelemetry-operator/main.tf

```diff
@@ -1,17 +1,33 @@
-module "operator" {
-  source        = "../helm-addon"
-  helm_config   = local.helm_config
-  irsa_config   = null
+module "cert_manager" {
+  source        = "../cert-manager"
   addon_context = var.addon_context
-
-  depends_on = [kubernetes_namespace_v1.prometheus]
 }
 
-resource "kubernetes_namespace_v1" "prometheus" {
+resource "kubernetes_namespace_v1" "adot" {
   metadata {
     name = local.helm_config["namespace"]
     labels = {
       "app.kubernetes.io/managed-by" = "terraform-aws-eks-blueprints"
+      "control-plane"                = "controller-manager"
     }
   }
 }
+
+data "kubectl_path_documents" "adot" {
+  pattern = "${path.module}/manifests/*.yaml"
+}
+
+# official kubernetes_manifest issue
+# https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest#before-you-use-this-resource
+resource "kubectl_manifest" "adot" {
+  for_each   = toset(data.kubectl_path_documents.adot.documents)
+  yaml_body  = each.value
+  depends_on = [kubernetes_namespace_v1.adot]
+}
+
+resource "aws_eks_addon" "adot" {
```

Comment:
Refer to the current structure for [CoreDNS](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/aws-coredns/main.tf#L5-L26) so we match that same level of functionality.